internal/daemon/api_files.go

```diff
@@ -483,18 +488,29 @@ func writeFile(item writeFilesItem, source io.Reader) error {
 	if err != nil {
 		return err
 	}
-	uid, gid, err := normalizeUidGid(item.UserID, item.GroupID, item.User, item.Group)
-	if err != nil {
-		return fmt.Errorf("cannot look up user and group: %w", err)
-	}
 	sysUid, sysGid := sys.UserID(osutil.NoChown), sys.GroupID(osutil.NoChown)
 	if uid != nil && gid != nil {
 		sysUid, sysGid = sys.UserID(*uid), sys.GroupID(*gid)
 	}
-
 	return atomicWriteChown(item.Path, source, perm, osutil.AtomicWriteChmod, sysUid, sysGid)
 }
 
+func mkdirAllUserGroup(path string, perm os.FileMode, uid, gid *int) error {
+	if uid != nil && gid != nil {
+		return mkdirAllChown(path, perm, sys.UserID(*uid), sys.GroupID(*gid))
+	} else {
+		return os.MkdirAll(path, perm)
```

Comment:
just call mkdirAllChown with osutil.NoChown that way the directory create is always within the mutex

Reply:
Done